### PR TITLE
Makes Portable scrubbers and atmos pumps repairable.

### PR DESCRIPTION
**Pull request by EnterTheJake**
## About The Pull Request

Atmos scrubbers and atmos pumps are now repairable.

## Why It's Good For The Game

Quality of life for atmos techs, You can repair your scrubbers and portable pumps now. 

## Changelog

:cl:

qol: You can now repair portable scrubbers and pumps.
/:cl:

### CODE REVIEW

**Review by SyncIt21 — CHANGES_REQUESTED** [review]
Add an examine and screentip and you should be good

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 1]
```diff
@@ -63,6 +63,21 @@
 	scrub(target.return_air())
 	return ..()
 
+/obj/machinery/portable_atmospherics/scrubber/welder_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(atom_integrity >= max_integrity || (machine_stat & BROKEN) || !tool.tool_start_check(user, amount = 1))
+		return ITEM_INTERACT_SUCCESS
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/portable/scrubber.dm:70]:
```suggestion
		return ITEM_INTERACT_BLOCKING
```

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 2]
```diff
@@ -63,6 +63,21 @@
...
+	to_chat(user, span_notice("You begin repairing [src]..."))
+	while(tool.use_tool(src, user, 2.5 SECONDS, volume=40))
+		atom_integrity = min(atom_integrity + 25, max_integrity)
+		if(atom_integrity >= max_integrity)
+			to_chat(user, span_notice("You've finished repairing [src]."))
+			return ITEM_INTERACT_SUCCESS
+		to_chat(user, span_notice("You repair some of the damage in [src]..."))
+
+	return ITEM_INTERACT_BLOCKING
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/portable/scrubber.dm:79]:
```suggestion
	return ITEM_INTERACT_SUCCESS
```

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 3]
```diff
@@ -63,6 +63,21 @@
...
+		return
+	if(atom_integrity >= max_integrity || (machine_stat & BROKEN) || !tool.tool_start_check(user, amount = 1))
+		return ITEM_INTERACT_SUCCESS
+	to_chat(user, span_notice("You begin repairing [src]..."))
+	while(tool.use_tool(src, user, 2.5 SECONDS, volume=40))
+		atom_integrity = min(atom_integrity + 25, max_integrity)
+		if(atom_integrity >= max_integrity)
+			to_chat(user, span_notice("You've finished repairing [src]."))
+			return ITEM_INTERACT_SUCCESS
+		to_chat(user, span_notice("You repair some of the damage in [src]..."))
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/portable/scrubber.dm:77]: Should use balloon alerts

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 4]
```diff
@@ -63,6 +63,21 @@
 	scrub(target.return_air())
 	return ..()
 
+/obj/machinery/portable_atmospherics/scrubber/welder_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/portable/scrubber.dm:68]:
```suggestion
		return NONE 
```
or
```suggestion
		return ITEM_INTERACT_SKIP_TO_ATTACK
```

File: code/modules/atmospherics/machinery/portable/scrubber.dm [hunk 5]
```diff
@@ -63,6 +63,21 @@
...
+/obj/machinery/portable_atmospherics/scrubber/welder_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return ITEM_INTERACT_SKIP_TO_ATTACK
+	if(atom_integrity >= max_integrity || (machine_stat & BROKEN) || !tool.tool_start_check(user, amount = 1))
+		return ITEM_INTERACT_BLOCKING
+	balloon_alert(user, "repairing...")
+	while(tool.use_tool(src, user, 2.5 SECONDS, volume=40))
+		atom_integrity = min(atom_integrity + 25, max_integrity)
+		if(atom_integrity >= max_integrity)
+			balloon_alert(user, "repaired!")
```
Comment (MrMelbert) [inline code/modules/atmospherics/machinery/portable/scrubber.dm:75]:
```suggestion
			balloon_alert(user, "repaired")
```

**Review by MrMelbert — COMMENTED** [review]
It should have screentips and/or examine text yes

**Comment by EnterTheJake** [comment]
Having quite a few integration issues, can't seem to know why.

**Pull request merged**